FurkannOzbek/application-hub
# [FEATURE/FIX] Homepage UI update & Conditional landing page button

# Pull Request Title

[FEATURE] Add Conditional Homepage Button and Styling Updates

---

## Description

This PR introduces a conditional homepage button that dynamically displays different options based on the user's login status. If the user is logged in, a "Dashboard" button is shown. If the user is not logged in, a "Sign Up" button is displayed. Additionally, the WhyChooseCard and Overview components have been styled, and a parallax image has been added to enhance the UI.

 - What problem does this PR solve?
   - It improves the user experience by tailoring the homepage to reflect the user's login state and updates the visual aesthetics of key components.
- What is the impact of the change?
  - Personalized homepage buttons improve user interaction.
  - Updated component styling enhances the overall design and usability.
  - The parallax image adds a modern and engaging visual element.


---

## Changes Made


1. **Frontend:**
 - Conditional rendering of the homepage button:
    - Shows "Dashboard" if the user is logged in.
    - Shows "Sign Up" if the user is not logged in.
 - Styling updates for:
    - WhyChooseCard component.
    - Overview component.
Added a parallax image for visual enhancement.
2. **Backend:** NA

---

## Screenshots (if applicable)

Desktop view : 
![UpdatedHomepageGif](https://github.com/user-attachments/assets/170c2c28-f1bd-4f80-9988-9bd4cdf30db8)


Mobile view:

https://github.com/user-attachments/assets/54355dc5-3d91-4685-9268-81d346f44b30

Landing Page Button If logged in : 
![loggedin - button change on homepage 1](https://github.com/user-attachments/assets/a4d87e7c-d683-4a09-8d64-f3dba64de75a)

Landing Page Button If not logged in : 

![not loggedin homepage button](https://github.com/user-attachments/assets/563b5da4-6588-4beb-a25a-a5159a57b7c2)


---

## Testing Steps

- Frontend:
  - Conditional Button:
     - Log in and verify that the "Dashboard" button is displayed.
     - Log out and confirm that the "Sign Up" button is shown.
  - Styling:
     - Inspect the WhyChooseCard and Overview components for consistent styling across different screen sizes.
  - Parallax Image:
     - Scroll through the page and confirm the parallax image behaves as expected.
